File: Data Structures & Algorithms/validate-parentheses/submission-5.java
class Solution {
    public boolean isValid(String str) {
        if (str == "" || str.length() % 2 == 1) {
            return false;
        }
        Stack<Character> s = new Stack<>();
        HashMap<Character, Character> m = new HashMap<>();
        m.put('}','{');
        m.put(']','[');
        m.put(')','(');
        for (int i = 0; i < str.length(); i++) {
            if (str.charAt(i) == '}' || str.charAt(i) == ']' || str.charAt(i) == ')') {
                if (!s.isEmpty() && s.peek().equals(m.get(str.charAt(i)))) {
                    s.pop();
                } else {
                    return false;
                }
            } else {
                s.push(str.charAt(i));
            }
        }
        return s.isEmpty();
    
    }
}

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-5.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        st = list()
        for i in range(0, len(tokens)):
            if tokens[i] == '+':
                st.append(int(st.pop()) + int(st.pop()))
            elif tokens[i] == '-':
                a = int(st.pop())
                b = int(st.pop())
                st.append(b-a)
            elif tokens[i] == '*':
                st.append(int(st.pop()) * int(st.pop()))
            elif tokens[i] == '/':
                a = int(st.pop())
                b = int(st.pop())
                st.append(b/a)
            else:
                st.append(tokens[i])
        return int(st.pop())